src/hotspot/share/opto/loopnode.hpp

```diff
@@ -1266,26 +1270,91 @@ class PhaseIdealLoop : public PhaseTransform {
   // Per-Node transform
   virtual Node* transform(Node* n) { return nullptr; }
 
-  Node* loop_exit_control(Node* x, IdealLoopTree* loop);
-  Node* loop_exit_test(Node* back_control, IdealLoopTree* loop, Node*& incr, Node*& limit, BoolTest::mask& bt, float& cl_prob);
-  Node* loop_iv_incr(Node* incr, Node* x, IdealLoopTree* loop, Node*& phi_incr);
-  Node* loop_iv_stride(Node* incr, IdealLoopTree* loop, Node*& xphi);
-  PhiNode* loop_iv_phi(Node* xphi, Node* phi_incr, Node* x, IdealLoopTree* loop);
+  Node* loop_exit_control(const Node* head, const IdealLoopTree* loop) const;
+
+  struct LoopExitTest {
+    Node* cmp = nullptr;
```

Comment:
I think it might be a good idea to use a more specific node type if you know before hand what kind of node the field will point to - in this case CmpNode*